### PR TITLE
Move API documentation from README to Accumulo.java

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/Accumulo.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ */
+package org.apache.accumulo.core.client;
+
+import org.apache.accumulo.core.client.impl.AccumuloClientImpl;
+import org.apache.accumulo.core.client.lexicoder.Lexicoder;
+import org.apache.accumulo.core.client.rfile.RFile;
+
+/**
+ * Entry point for majority of Accumulo's public API. Other Accumulo API entry points are linked
+ * later.
```
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/Accumulo.java]: Later? Could say linked below.

**Comment by joshelser** [comment]
Doesn't this make it a bit harder for the average user to know what public API is? I wouldn't have thought we had a problem with a sprawling public API (actually would've said we're pretty darn good at a minimal one).

**Comment by ctubbsii** [comment]
I prefer the public API being declared in the README, not the Javadoc. Javadoc is a pain to maintain, and I think its use should be limited to describing API semantics of the specific, narrowly-scoped class, method, field, or even package it is attached to. Further, it is intended to be viewed in a rendered form... not raw HTML embedded inside a Java comment. Linking from the README to the Accumulo.java file isn't doing users any favors.

Also, I'm pretty sure there's no reason to turn off checkstyle for the javadoc. That's just asking for trouble, potentially introducing rendering issues that should have been caught earlier.

As for the class move, I think that's fine, although ultimately, I'd prefer the main entry point be in a separate API jar, so the API definition is just "everything accessible in accumulo-api.jar", but that will take more work to avoid dependency cycles between jars.

**Comment by ctubbsii** [comment]
I thought of another reason why the README pointing to the Accumulo.java file is... the README is also for consumers of the binary tarball... which does not contain the Accumulo.java file. If we move the public API information to the javadoc, we'd have to have a different README for the binary and source tarballs.

At this point, I think the case is pretty strong to keep the public API statement in the README. @keith-turner What do you think?

**Comment by keith-turner** [comment]
> Doesn't this make it a bit harder for the average user to know what public API is?

Possibly, its hard to know w/o data.   I am not convinced the readme is better place for new devs to find this critical piece of information.  My thinking was that javadoc is something that developers read in their IDE, therefore its important we define the API there.  I had two goals for this javadoc that I thought new developers were likely to read:

  * Define the public API
  * Help new developers find other top level API entry points 

>  the README is also for consumers of the binary tarball... which does not contain the Accumulo.java file

Thats a good point.   Based on these comments I think the following would be a better solution than defining the API in Readme.md or Accumulo.java.

  * Define the API on the website at a place where it has a stable link, like https://accumulo.apache.org/api.   
  * Link to this definition from README.md, Accumulo.java, Accumulo Tour, Accumulo User manual, and any other docs written for developers.  

I took a quick look at the tour and the user manual section on writing clients and did not see mention of the API.  We could make all of these places link to the README on github, but I think linking to https://accumulo.apache.org/api is more elegant.  This way we can avoid trying to decide the one place that would be most beneficial to place this info, instead link to it from any documentation written for new developers.

**Comment by milleruntime** [comment]
> I took a quick look at the tour and the user manual section on writing clients and did not see mention of the API. We could make all of these places link to the README on github, but I think linking to https://accumulo.apache.org/api is more elegant.

I like this a lot.  I found it difficult when writing the tour to have a simple way of explaining what is API and what is not.  This way, we could just link to the API web page.

I think a single web page devoted to the API would be perfect. We can link to it from the javadoc and readme and even have a link on the main page.

**Comment by keith-turner** [comment]
> We can link to it from the javadoc and readme and even have a link on the main page.

I think a link on the main page is ok as long as it does not clutter.   Even if there is a link on the main page, I still think explicit links in docs for new devs are important to guide new users to this important info.

**Comment by ctubbsii** [comment]
I thought the README was a pretty stable location already, but obviously, it's limited in the amount of details we should put there so it doesn't get bloated. A dedicated central place on the website is fine with me.

**Comment by joshelser** [comment]
> A dedicated central place on the website is fine with me.

+1, as long as it's "human readable" and prominent, I think it's fine. My experience is that folks do not venture into Javadoc unless they have a very specific question.

**Comment by keith-turner** [comment]
> A dedicated central place on the website is fine with me.

I can create PRs for the website and code to make this change.  I will leave this PR open until subsequent PRs or issues are created.  I'll hold off on working on it until @joshelser has had a chance to respond to this proposed change.

**Comment by keith-turner** [comment]
Just saw the comment from @joshelser after posting.  I'll get started on this.

**Comment by keith-turner** [comment]
Opened #660 and apache/accumulo-website#109

**Pull request closed without merging**